ipn/ipnlocal/serve.go

```diff
@@ -214,13 +219,17 @@ func (b *LocalBackend) updateServeTCPPortNetMapAddrListenersLocked(ports []uint1
 }
 
 // SetServeConfig establishes or replaces the current serve config.
-func (b *LocalBackend) SetServeConfig(config *ipn.ServeConfig) error {
+// ETag is an optional parameter to enforce Optimistic Concurrency Control.
+// It is a pointer to a string to ensure backwards compatibility with clients
```

Comment:
Pointer to string is pretty weird. Are you implicitly saying here that nil means don't use an etag? Then why can't an empty string mean the same?

Reply:
Given the changes below, this is no longer needed. Further explanation below for brevity:

Initial intention was to distinguish between clients who don't send etags at all, or might send an empty string one (the initial state of the `serveConfigETag` field. But with your suggestion, `etag` is never an empty string, because even if the serve config is `nil`, then json-ifying that gives you `"null"` which gives a non-empty string sha256.